Recuperação de senha

Sckyz opened the pull request.

Sckyz committed:
🚧 Recuperar senha em progresso

Sckyz committed:
Merge branch 'develop' into feature/change-password

vitorcorreasx committed:
🚑️ Ajuste geral nos modais para o crud

Sckyz committed:
✨ Criada rotas e páginas para recuperação de senha

Sckyz committed:
🩹 Acento que não existe

nos prints do pr tá errado pq vi depois o erro

Sckyz committed:
🎨 Criada computed para as rotas que não mostraram a tabHeader

Sckyz committed:
⚡️ Arrumada as rotas

Sckyz committed:
Update notifications.ts

Sckyz committed:
🎨 Correções

julianostroschon commented on locales/br/notifications.ts:
```diff
@@ -28,5 +28,9 @@ export default {
       notSelected: "Você deve selecionar pelo menos um sistema para deletar",
     },
+    emptyField: "Algum campo está vazio | O campo está vazio",
+    invalidEmail: "Não é um email válido",
```
email inválido

Sckyz replied:
Troquei "Não é um email válido" para "Email inválido"

julianostroschon commented on locales/br/notifications.ts:
```diff
@@ -28,5 +28,9 @@ export default {
         "Houve um erro ao deletar o sistema | Houve um erro ao deletar os sistemas",
       notSelected: "Você deve selecionar pelo menos um sistema para deletar",
     },
+    emptyField: "Algum campo está vazio | O campo está vazio",
```
uso de plural pra isso?

Sckyz replied:
Uso incorreto de pluralização, agora está arrrumado

julianostroschon commented on src/components/DynamicDialog/SystemModals/AddSystemDialog.vue:
```diff
@@ -1,33 +1,33 @@
   sublabel: "",
   link: "",
-  sistema: props.systemType,
+  sistema: props.item[0].sistema,
```
pq a posição 0?

Sckyz replied:
pergunta pro vitor

Sckyz replied:
eu nao consigo editar esse arquivo, diz que é só de leitura

Sckyz replied:
Pois estava pegando o primeiro index do array para deixar como default, se voce fosse criar um sistema em gestão, "gestão" ja estária no input do modal, mesmo serve para sistema rh. Porém agora ja arrumei e esta de outra forma

julianostroschon commented on src/components/DynamicDialog/SystemModals/AddSystemDialog.vue:
```diff
@@ -1,33 +1,33 @@
     await runMutation(AddSystem, { data: form });
     positiveNotify(t("notifications.success.createSystem"));
+    emits("cancel");
+    emits("reload");
```
emissão de 2 eventos, desnecessário

Sckyz replied:
ja foi resolvido pelo vitor

julianostroschon commented on src/components/DynamicDialog/SystemModals/AddSystemDialog.vue:
```diff
@@ -41,36 +41,49 @@ async function createSystem() {
-      </div>
+    <div class="row q-pa-xl">
+      <q-input
+        class="col-6 q-px-xs"
+        v-model="form.label"
+        :label="$t('text.title')"
+        :rules="[(val: string) => validateNotEmpty(val)]"
+      />
+      <q-input
+        class="col-6 q-px-xs"
+        v-model="form.icon"
+        label="Icone"
+        :rules="[(val: string) => validateNotEmpty(val)]"
+      >
```
validação co,m a vee-validate

Sckyz replied:
feito

Sckyz replied:
feito

julianostroschon commented on src/components/DynamicDialog/SystemModals/EditSystemDialog.vue:
```diff
@@ -0,0 +1,152 @@
+    });
+    positiveNotify(t("notifications.success.editSystem"));
+    emits("cancel");
+    emits("reload");
```
emissão de 2 eventos, desnecesário

Sckyz replied:
eu nao consigo editar esse arquivo, diz que é só de leitura

julianostroschon commented on src/components/DynamicDialog/SystemModals/EditSystemDialog.vue:
```diff
@@ -0,0 +1,152 @@
+    </div>
+
+    <div v-else class="row q-pa-xl">
+      <q-input
+        class="col-6 q-px-xs"
+        v-model="form.label"
+        :label="$t('text.title')"
+        :rules="[(val: string) => validateNotEmpty(val)]"
+      />
+      <q-input
+        class="col-6 q-px-xs"
+        v-model="form.icon"
+        label="Icone"
+        :rules="[(val: string) => validateNotEmpty(val)]"
+      >
```
validação

Sckyz replied:
eu nao consigo editar esse arquivo, diz que é só de leitura

Sckyz replied:
pronto, feita a validação com vee-validate + yup

julianostroschon commented on src/components/Login/ChangePassword.vue:
```diff
@@ -0,0 +1,94 @@
+const passwordPattern = /^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)[A-Za-z\d]{4,8}$/;
+
+const validatePassword = () => {
+  if (emptyFieldPattern.test(form.newPwd && form.confPwd)) {
+    errorMessage.value = t("notifications.fail.emptyField");
+    return;
+  }
+  if (!passwordPattern.test(form.newPwd)) {
+    errorMessage.value = t("notifications.fail.invalidPassword");
+    return;
+  }
+  if (form.newPwd !== form.confPwd) {
+    errorMessage.value = t("notifications.fail.equalPassword");
+    return;
+  }
```
validação

Sckyz replied:
feita validação com vee-validate

Sckyz committed:
🚧 Criando mutation de recuperar senha

Sckyz committed:
Merge branch 'develop' into feature/change-password

julianostroschon commented on src/components/SistemaGestao/SistemaGestao.vue:
```diff
@@ -71,6 +75,12 @@ async function getGestaoList() {
 }
+
+const buttons: Action[] = [
+  { label: "Adicionar", icon: "add", component: AddSystemDialog },
```
tradução

Sckyz replied:
tradução feita

julianostroschon commented on src/components/SistemaRh/SistemaRh.vue:
```diff
@@ -70,6 +74,12 @@ async function getListRh() {
 }
+
+const buttons: Action[] = [
+  { label: "Adicionar", icon: "add", component: AddSystemDialog },
```
tradução

Sckyz replied:
feita tradução destes botões

julianostroschon reviewed (changes requested):
uns comentários

Sckyz committed:
🌐 Novas traduções

Sckyz committed:
🦺 Validação para o email, username e password

Sckyz committed:
🚧 Telas login, mandar email e mudar senha

Sckyz committed:
🚧 Mutation mandar email em andamento

Sckyz committed:
🚧 .

Sckyz committed:
🌐 Ajustes nas traduções

- Email nao encontrado e validações (logo serão adicionadas mais traduções sobre validação)

Sckyz committed:
🦺 Ajuste na estrutura do schema yup

Criação de constantes que podem ser reusadas em vários schemas

Sckyz committed:
⚰️ Tela sem uso

Apenas uma notificação positiva ou negativa já serve

Sckyz committed:
🎨 Ajustes tanto na validação quanto na mutation de enviar email para recuperação de senha

Sckyz committed:
🌐 Tradução de botões

Sckyz committed:
🦺 Ajustes na validação do login

Sckyz committed:
🦺 Ajustes por conta do login

O login não deveria usar o userSchema que usamos para cadastrar novos usuários, e sim um próprio pra ele que válida

Sckyz commented:
![image](https://github.com/partithura-estagiarios/frontend-intranet-elyte/assets/122920355/cd8e9f3e-5786-452d-88b5-14f8aba18acc)


eu cortei sem terminar de escrever hahah,.... espaços em branco

Sckyz committed:
🙈 Update

Sckyz committed:
🦺 Criada validação para os inputs dos modais

Sckyz committed:
🎨 Removido código "sistema: props.item[0].sistema"

Agora o sistema é por props

Sckyz committed:
🦺 Feita validaação com vee-validate e yup

Sckyz committed:
🎨 Ajustes por conta da validação

julianostroschon commented on src/components/SistemaGestao/SistemaGestao.vue:
```diff
@@ -12,6 +12,7 @@
             :buttons="actionButtons"
             :item="gestaoList"
             @reload="getGestaoList()"
+            system="gestão"
```
pq esse texto aqui?

julianostroschon commented on src/graphql/changePwd/ChangePwd.gql:
```diff
@@ -0,0 +1,5 @@
+mutation UpdatePassword($data: changePasswordInput!) {
```
tipos devem começar com letra maiúscula

Sckyz replied:
pronto, mudei para ChangePasswordInput

Sckyz committed:
🩹 Case do tipo arrumado
